### PR TITLE
Vineet Kumar Task6023_IndexNamesPlaceHolder

### PR DESCRIPTION
**Pull request by regnosys-prod-user**
Please see Issue  #2307

### CODE REVIEW

File: rosetta-source/src/main/rosetta/product-asset-type.rosetta [hunk 1]
```diff
@@ -52,20 +52,22 @@ type CreditDefaultPayout extends PayoutBase: <" The credit default payout specif
 		priceQuantity exists
 
 type GeneralTerms: <" A class specifying a set of non-monetary terms for the Credit Derivative Transaction, including the buyer and seller and selected items from the ISDA 2014 Credit Definition article II, such as the reference obligation and related terms. The CDM GeneralTerms class corresponds to the FpML GeneralTerms complex type, except that the effectiveDate and scheduledTerminationDate have been positioned as part of the InterestRatePayout class in the CDM instead of in GeneralTerms.">
+	
+    referenceInformation ReferenceInformation (0..1) <"This attribute contains all the terms relevant to defining the reference entity and reference obligation(s).">
+    indexReferenceInformation IndexReferenceInformation (0..1) <"This attribute contains all the terms relevant to the underlying Index.">
+    additionalIndexReferenceInformation AdditionalIndexReferenceInformation (0..1) <"This attribute contains all the additional terms relevant to defining the Credit DefaultSwap Index.">
```
Comment (iansloyan) [inline rosetta-source/src/main/rosetta/product-asset-type.rosetta:59]: why is indexReferenceInformation being explicitly added as an attribute? surely if an index is specified then that information will be inside "index"

File: rosetta-source/src/main/rosetta/observable-asset-type.rosetta [hunk 2]
```diff
@@ -331,6 +332,7 @@ type FloatingRateOption: <"Specification of a floating rate option as a floating
 	inflationRateIndex InflationRateIndexEnum (0..1) <"The reference index that is used to specify the inflation interest rate. The FpML standard maintains the list of such indices, which are positioned as enumeration values as part of the CDM.">
 		[metadata scheme]
 	indexTenor Period (0..1) <"The ISDA Designated Maturity, i.e. the floating rate tenor.">
+    indexReferenceInformation IndexReferenceInformation (0..1) <"">
```
[the remark this answers is not in the document]
Comment (winitall01) [reply rosetta-source/src/main/rosetta/observable-asset-type.rosetta:336]: This is based on one of the requirements for Floating Rate Legs names. this is explained in detail here
https://github.com/finos/common-domain-model/issues/2307

File: rosetta-source/src/main/rosetta/product-asset-type.rosetta (hunk 1 above)
Comment (winitall01) [reply rosetta-source/src/main/rosetta/product-asset-type.rosetta:59]: 'indexReferenceInformation' is already present here. In this change, the 'indexReferenceInformation' in split into 'indexReferenceInformation' and 'additionalIndexReferenceInformation'. 
We've just added 'additionalIndexReferenceInformation' in the type above.

File: rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta [hunk 3]
```diff
@@ -74,6 +76,8 @@ type ProductTaxonomy extends Taxonomy: <"Specifies the product taxonomy, which i
 		optional choice value, productQualifier
 
 type Index extends ProductBase: <"Identifies an index by referencing a product identifier.">
+    indexReferenceInformation IndexReferenceInformation (0..1) <"This attribute contains all the terms relevant to defining the Credit DefaultSwap Index.">
+
```
[the remark this answers is not in the document]
Comment (winitall01) [reply rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta]: This is corrected now. Thanks for point it @iansloyan

**Comment by PayalKhanna** [comment]
replaced with https://github.com/finos/common-domain-model/pull/2318

**Comment by lolabeis** [comment]
The proposal's general idea looks fine.

There are a few implementation points that should be adjusted as follows:
- The simplified 'IndexReferenceInformation` should be moved from `product.asset` to `base.staticdata.asset.common`. This will avoid the awkward namespace import at the top of the latter (`product` is meant to import from `base`, not the other way round).
- Rather than having a separate `AdditionalIndexReferenceInformation` type, put the additional attributes into a new `CreditIndexReferenceInformation` type that extends `IndexReferenceInformation`, and encapsulate only that new type as an attribute in `GeneralTerms` (instead of encapsulating as 2 separate attributes).
- Note sure why `indexAnnexSource` has been kept inside the simplified `IndexReferenceInformation`, this seems to belong with the additional attributes.

**Comment by PayalKhanna** [comment]
replaced with https://github.com/finos/common-domain-model/pull/2335

**Pull request closed without merging**